src/types/rest/response/private-copy-trading.ts

```diff
@@ -0,0 +1,305 @@
+  nickName: string;
+  uniqueCode: string;
+  copyNum: string;
+  copyTotalAmt: string;
+  copyTotalPnl: string;
+  copyAmt: string;
+  copyMode: 'fixed_amount' | 'ratio_copy';
+  copyRatio: string;
+  ccy: string;
+  profitSharingRatio: string;
+  beginCopyTime: string;
+  endCopyTime: string;
+  copyRelId: string;
+  copyState: '0' | '1';
+  leadMode: 'public' | 'private';
+}
+
+export interface GetCopyTradingConfigResult {
+  maxCopyAmt: string;
+  minCopyAmt: string;
+  maxCopyTotalAmt: string;
+  minCopyRatio: string;
+  maxCopyRatio: string;
+  maxTpRatio: string;
+  maxSlRatio: string;
+}
+
+export interface PnlRatio {
+  beginTs: string;
+  pnlRatio: string;
+}
+
+export interface LeadTraderRank {
+  accCopyTraderNum: string;
+  aum: string;
+  ccy: string;
+  copyState: '0' | '1';
+  copyTraderNum: string;
+  leadDays: string;
+  maxCopyTraderNum: string;
+  nickName: string;
+  pnl: string;
+  pnlRatio: string;
+  pnlRatios: PnlRatio[];
+  portLink: string;
+  traderInsts: string[];
+  uniqueCode: string;
+  winRatio: string;
+}
+
+export interface GetLeadTraderRanksResult {
+  dataVer: string;
+  ranks: LeadTraderRank[];
+  totalPage: string;
+}
+
+export interface LeadTraderPnl {
+  beginTs: string;
+  pnl: string;
+  pnlRatio: string;
+}
+
+export interface LeadTraderStats {
+  avgSubPosNotional: string;
+  ccy: string;
+  curCopyTraderPnl: string;
+  investAmt: string;
+  lossDays: string;
+  profitDays: string;
+  winRatio: string;
+}
+
+export interface LeadTraderPreference {
+  ccy: string;
+  ratio: string;
+}
+
+export interface LeadTraderCurrentPosition {
+  ccy: string;
+  instId: string;
+  instType: string;
+  lever: string;
+  margin: string;
+  markPx: string;
+  mgnMode: string;
+  openAvgPx: string;
+  openTime: string;
+  posSide: string;
+  subPos: string;
+  subPosId: string;
+  uniqueCode: string;
+  upl: string;
+  uplRatio: string;
+}
+
+export interface LeadTraderPositionHistory {
+  ccy: string;
+  closeAvgPx: string;
+  closeTime: string;
+  instId: string;
+  instType: string;
+  lever: string;
+  margin: string;
+  mgnMode: string;
+  openAvgPx: string;
+  openTime: string;
+  pnl: string;
+  pnlRatio: string;
+  posSide: string;
+  subPos: string;
+  subPosId: string;
+  uniqueCode: string;
+}
+
+export interface CopyTraderInfo {
+  beginCopyTime: string;
+  nickName: string;
+  portLink: string;
+  pnl: string;
+}
+
+export interface GetCopyTradersResult {
+  copyTotalPnl: string;
+  ccy: string;
+  copyTraderNumChg: string;
+  copyTraderNumChgRatio: string;
+  copyTraders: CopyTraderInfo[];
+}
+
+export interface GetPrivateLeadTraderRanksRequest {
+  instType?: 'SWAP';
+  sortType?:
+    | 'overview'
+    | 'pnl'
+    | 'aum'
+    | 'win_ratio'
+    | 'pnl_ratio'
+    | 'current_copy_trader_pnl';
+  state?: '0' | '1';
+  minLeadDays?: '1' | '2' | '3' | '4';
+  minAssets?: string;
+  maxAssets?: string;
+  minAum?: string;
+  maxAum?: string;
+  dataVer?: string;
+  page?: string;
+  limit?: string;
+}
```

Comment:
Should be in the requests subdir. Doesn't seem like there's one for copy trading yet, so feel free to make a new file here:
https://github.com/tiagosiebler/okx-api/tree/master/src/types/rest/request

Don't forget to add the export for the new file to the index.ts in this folder

Reply:
There is a file already, requests just dont have private/public names.